packages/lesswrong/lib/utils/schemaUtils.ts

```diff
@@ -64,7 +64,12 @@ const generateIdResolverMulti = <CollectionName extends CollectionNameString>({
 // If the user can't access the document, returns null. If the user can access the
 // document, return a copy of the document in which any fields the user can't access
 // have been removed. If document is null, returns null.
-export const accessFilterSingle = async <T extends DbObject>(currentUser: DbUser|null, collection: CollectionBase<T>, document: T|null, context: ResolverContext|null): Promise<T|null> => {
+export const accessFilterSingle = async <N extends CollectionNameString>(
+  currentUser: DbUser|null,
+  collection: CollectionBase<N>,
+  document: ObjectsByCollectionName[N]|null,
+  context: ResolverContext|null,
+): Promise<Partial<ObjectsByCollectionName[N]>|null> => {
```

Comment:
I predict this `Partial` thing becoming a problem (because it will make tons of fields maybe-null when really they can be read by anyone), we had this problem on TAIGA (same comment about the other `Partial`s in this file)

Reply:
I bet we can solve this by fixing our code-gen such that generated fragment types take into account field permissions (any field with `guest` permissions is left alone, any field with stricter permissions has an `| null` appended to it, if it isn't already nullable).  We've in any case had this as a source of bugs in the past, where otherwise non-nullable fields had field permissions that implied they wouldn't always be available on the client, but the types didn't reflect this.

Reply:
Yeah - I'm not 100% convinced by this, but I think it's better than the status quo. Probably more work should be done here in the future.